### ISSUE TITLE
How to inform users that auto-sync is activated

### ISSUE THREAD

**Issue by dmtrjsg** [Joystream/atlas#3449]
## Context

Up until this point in time we've always been targeting a release with manual video upload. It becomes apparent now that we are nearing completion of auto-sync service so will be launched together with autosync. 

We need to think about the effective ways to communicate this to the creators, as right now there is nothing in the app apart from the Value Proposition on the landing page. 

## Scope
- [ ] YPP sync active status
- [ ] Explainer modal that it is on?

Important: 
Before implementing this pls consider: 
There's an Opt out feature from the YPP project which we 
- https://github.com/Joystream/atlas/issues/3450
- https://github.com/Joystream/youtube-synch/issues/45

**Comment by toiletgranny** [comment]
@dmtrjsg, a few follow-up questions before I get started:
1. Is this technically possible for me, as a channel owner, to change the video category to which my newly uploaded videos should be assigned going forward? With no effect on the videos that were already carried over to Joystream at this point? If so, are we open to the idea of adding such control somewhere in the UI in the Studio?
2. I'm assuming that details of the auto-sync mechanism will be explained in the Terms & conditions step of the Authorize channel dialog. But will there be some other place (such as the JS Handbook or something else) for me, as a channel owner, to get back to this later? If not, are we open to the idea of explaining it somewhere in the UI in the Studio? 
   - The reason I'm asking about this is that as a video creator, I might be super careful about making sure I understand how video settings are mapped between Atlas and YouTube. By video settings, I mean things such as video visibility (I can imagine I wouldn't like my YouTube unlisted video to become public on Atlas, right?), license (also potentially a susceptible subject), video language, comment section status, subtitles, and more.

**Comment by toiletgranny** [comment]
Here's how Odysee does that: https://help.odysee.tv/category-syncprogram/. Basically, rather than going for extensive in-app explanations, they just give you the essential information and leave a link to this page if you're hungry for more.

Their limits page is worth checking out, too: https://help.odysee.tv/category-syncprogram/limits/

**Comment by dmtrjsg** [comment]
@toiletgranny good questions on the category actually..  at the moment this is not possible to change it, once selected, but I think it may make sense to at least display it as read only somewhere. If requested by some of the creators we would add this feature. 

Re auto-sync info, that is only contained on the T&Cs page. I think we could add a Checkbox "[ ] Opt in to auto-sync service, to upload my Youtube channel videos to Gleev." But equally I believe it would make sense to show the auto-sync status on the dashboard. 

Showing which category is selected prob would make sense somewhere.

On the additional info, yes the link to Notion from the dasboard will get creators to the page where all programme details are described in detail.

**Comment by toiletgranny** [comment]
@dmtrjsg my recommendations for displaying the status of the auto-sync feature, along with a few more suggestions (made in the context of the recent bug bash and the issues you've been raising in the last couple of days), are ready to be reviewed. Find out all about it from:

- [Loom video update](https://www.loom.com/share/658d47206b44411fb546d011a8d2b165) (~11 min at 1.5x speed),
- [Figma prototype](https://www.figma.com/proto/DiQrntuI4KlEdudZdlsH3f/How-to-inform-users-that-auto-sync-is-activated?page-id=1519%3A26568&node-id=1519%3A26800&viewport=281%2C192%2C0.08&scaling=scale-down&starting-point-node-id=1519%3A26816&show-proto-sidebar=1).

I will update the final design files for YPP once this is approved.

**Comment by dmtrjsg** [comment]
Excellent work ⭐ 

1. Settings tab seems like a good way to house all of the info 
2. For licenses there are only 2 on YT, standard-youtube and CC. For the former we decided to create Joystream-standard license and will be mapping to identical options. 
3. Additional settings such as comments enabled and mature content is a little early to work on but I like your thinking. Let's however keep it out of scope for now before we start having creators onboaraded. 
4. Re YT sync by default, I think we can be more agressive with offering this option. E.g. by default it's ON for everyone but they can disable it in the settings after they sign up. So we just need to inform them on this step. ofc its always a trade-off between the drop off on this step and having too little subscribers actually opting in after sign up. I absolutely love the separate step of the journey for YT sync info and category selector. RE license we can just say that YT-standard will be linked to Joysatream standard, and CC to CC0.

**Comment by toiletgranny** [comment]
Ok, great, thanks, @dmtrjsg! To confirm, do I have a green light to add this step to the YPP sign-up flow? Under the condition that it's set to "Import my YouTube videos" by default?
<img width="320" alt="image" src="https://user-images.githubusercontent.com/35307309/208126621-8dedb801-0e88-4168-ae75-a4a8960a3eec.png">

A few more questions irt opting out from the YPP entirely to help me make sure the copy is accurate:
1. If I opt out of the program altogether, will I be able to opt-in back again using the same Joystream channel?
2. Same question, but with a different Joystream channel under the same Joystream member?
3. Same question, but both with a different Joystream channel AND a different Joystream membership?
4. If the answer to all of the above is YES, will I be rewarded for joining the program every time? Or does it work only the first time a YouTube channel is associated with a Joystream channel?

**Comment by dmtrjsg** [comment]
Extra step -  yes let's go for it in terms of communicating this - makes sense to have it. I'm however hesitant to give them the option to opt out at this step. I'd rather inform that by default its ON and they can opt out in the settings.  This is under assumption that we are aqcuiring creators with 100% crypto content, which will be largely the case so I don't see how this would result in polluting the platform with irrelevant content. 

Do you think we risk having drop offs at this stage if we don't offer simple option to top out right in the ob flow?

Answers: 
1. Yes can opt back in with same channel
2. You can run multiple JS channels from same JS member. YPP programme is rewarded on Channel level.
3. Ahh that's interesting, we should not and I will ask  Zeeshan if we prevent from this happening?
4. Reward for joining only the first time, and the provisions should have been made to prevent this re-qualifying for the sign up rewards. Achieved by keeping the original record in the DB once opted out and continuing with the same channel (just change flag to assigned) once assigned back in. 

@zeeshanakram3 can you pls comment on points 3 and 4?

**Comment by zeeshanakram3** [comment]
> 3. Ahh that's interesting, we should not and I will ask Zeeshan if we prevent from this happening?

We don't prevent this. Also, maybe I am misunderstanding something, but why we should prevent creator to link their YT channel with a  different Joystream channel AND a different Joystream membership from what they used in the previous YPP signup?

> 4. Reward for joining only the first time, and the provisions should have been made to prevent this re-qualifying for the sign up rewards. Achieved by keeping the original record in the DB once opted out and continuing with the same channel (just change flag to assigned) once assigned back in.

Sounds good. So adding some flag that tracks if this is creator's first signup or re-opt in, etc actually makes sense

**Comment by dmtrjsg** [comment]
> why we should prevent creator to link their YT channel with a different Joystream channel AND a different Joystream membership

This way someone can create multiple joystream channels feeding off single YT channel and benefit from X'ed rewards. 

> Adding some flag that tracks if this is creator's first signup or re-opt in, etc actually makes sense
issue raised
- https://github.com/Joystream/youtube-synch/issues/102

**Comment by traumschule** [comment]
> We need to think about the effective ways to communicate this to the creators, as right now there is nothing in the app apart from the Value Proposition on the landing page.

Over time GH added more and more items to the top navbar. Suddenly a new icon appeared with some sort of indicator that showed details on hover.

![gh-notification](https://user-images.githubusercontent.com/31551045/208198698-59fcb80c-7b34-4736-bd62-253d5fa6bebd.png)

If there's an icon for the YPP/sync feature which according to the discussion will be enabled by default, the indicators could be (colors just wild picks)
- `#369a31`: enabled and fully functional, nothing to do according to settings
  - click opens settings
- `#000bff`: syncing, uploads in progress
  - opens progress modal/tab/page
- `#ffed00`: warning, upload errors, user interaction required
  - (eg. to approve syncing of new yt videos)
  - click opens modal/page with details
  - (optionally  offers to change settings, like reducing required interactions)
- `#ff0000`: feature disabled
  - can notification icon be disabled in settings by opting out of the program completely?

**Comment by drillprop** [comment]
I don't have anything to add. I would go with the additional tab in the dashboard and the extra step in ypp sign-up flow. That's a green light from me as well.

**Comment by toiletgranny** [comment]
> Extra step - yes let's go for it in terms of communicating this - makes sense to have it. I'm however hesitant to give them the option to opt out at this step. I'd rather inform that by default its ON and they can opt out in the settings. This is under assumption that we are aqcuiring creators with 100% crypto content, which will be largely the case so I don't see how this would result in polluting the platform with irrelevant content.
> 
> Do you think we risk having drop offs at this stage if we don't offer simple option to top out right in the ob flow?

@dmtrjsg Yes, I think the risk is rather high. As a prolific creator, I might want to make sure I have a total understanding of how the auto-sync feature works before all my content (which might also be subject to sponsor contracts) is migrated to Joystream. If we don't offer an opt-in at this stage, it means I have to do all the heavy reading beforehand which ultimately decreases my chances of completing the flow. 

I don't think mentioning an option of turning the auto-sync off later works, as — at this stage — it's unclear to me, as a creator, where do I look for the switch, how fast can I find it, and how much of my content gets migrated before that happens.

My recommendation is to have the option to disable the auto-sync during the YPP sign-up flow.

--- 

Reiterating my initial four questions irt re-joining the YPP, having followed all the discussion here and, later, on Discord, let me try to recap on what I think the final answers:

1. _If I opt out of the program altogether, will I be able to opt-in back again using the same Joystream channel?_
   **Yes.**
2. _Same question, but with a different Joystream channel under the same Joystream member?_
   **Yes.**
3. _Same question, but both with a different Joystream channel AND a different Joystream membership?_
    **Yes.**
4. _If the answer to all of the above is YES, will I be rewarded for joining the program every time? Or does it work only the first time a YouTube channel is associated with a Joystream channel?_
   **No. A single YouTube channel can only be rewarded once for joining YPP. So, while this is technically possible for me to opt-out from a YPP, and opt-in back again using a different JS channel and membership, I won't receive a reward if the YT channel I used has already been associated with another JS channel in the past..**

@dmtrjsg can you confirm? 🙏 

--- 

@traumschule Thanks for the suggestion!

**Comment by dmtrjsg** [comment]
@toiletgranny 

Thanks for clarifying:

1. Ability to opt out at this stage -> let's go for it, but pls have opt-in by default and as a first option in the list 😇 
2. The round up of the answers 1-4 is correct 👍

**Comment by toiletgranny** [comment]
@dmtrjsg This is ready to be implemented! Designs, docs, and changelog [can be found here](https://www.figma.com/file/oQqFqdAiPu16eeE2aA5AD5/YouTube-Partner-Program?node-id=3338%3A84773&t=f1pw4rw8EhIsAFB3-4).
